server/src/controllers/BookingController.mjs

```diff
@@ -1,142 +1,82 @@
-
-function countCustomer(BookingID) {
-	return bookingDetailsData.filter(
-		(booking) => booking.BookingID === BookingID
-	).length;
-}
-
-export const resolveBookingById = (req, res) => {
-	const { id } = req.params;
-	if (isNaN(id)) {
-		return res
-			.status(StatusCodes.BAD_REQUEST)
-			.send("Invalid ID supplied");
-	}
-	const bookingIndex = bookingData.findIndex(
-		(booking) => booking.BookingID === parseInt(id)
-	);
-	if (bookingIndex === -1) {
-		return res
-			.status(StatusCodes.NOT_FOUND)
-			.send("Booking not found");
-	}
-
-	return bookingIndex;
-};
+import BookingModel from '../models/BookingModel.mjs';
+import customerModel from '../models/CustomerModel.mjs';
+import RoomModel from '../models/RoomModel.mjs';
+import BookingCustomerModel from '../models/BookingCustomerModel.mjs';
+import { StatusCodes } from 'http-status-codes';
 
 export const BookingController = {
-	// get all bookings
-	get: (req, res) => {
-		const { BookingID, RoomID, BookingDate, CheckOutDate, Cost } =
-			req.query;
-		let bookings = bookingData;
-
-		// check valid value
-		if (BookingID && isNaN(BookingID)) {
-			return res
-				.status(StatusCodes.BAD_REQUEST)
-				.send("Invalid BookingID supplied");
-		}
-		if (RoomID && isNaN(RoomID)) {
-			return res
-				.status(StatusCodes.BAD_REQUEST)
-				.send("Invalid RoomID supplied");
-		}
-		if (Cost && isNaN(Cost)) {
-			return res
-				.status(StatusCodes.BAD_REQUEST)
-				.send("Invalid Cost supplied");
-		}
-
-		// filter bookings
-		if (BookingID) {
-			bookings = bookings.filter(
-				(booking) =>
-					booking.BookingID ===
-					parseInt(BookingID)
-			);
-		}
-		if (RoomID) {
-			bookings = bookings.filter(
-				(booking) => booking.RoomID === parseInt(RoomID)
-			);
-		}
-		if (BookingDate) {
-			bookings = bookings.filter(
-				(booking) => booking.BookingDate === BookingDate
-			);
-		}
-		if (CheckOutDate) {
-			bookings = bookings.filter(
-				(booking) =>
-					booking.CheckOutDate === CheckOutDate
-			);
-		}
-		if (Cost) {
-			bookings = bookings.filter(
-				(booking) => booking.Cost === parseInt(Cost)
-			);
-		}
+  createNewBooking: async (req, res) => {
+    //req: {RoomID: , Customers: [{Name:, IdentityCard:, address:, Type:, }, {Name:, IdentityCard:, address:, Type:, }]}
+    try {
+      const { RoomId, Customers } = req.body;
+      const getCurrentDate = () => {
+        const now = new Date();
+        const year = now.getUTCFullYear();
+        const month = String(now.getUTCMonth() + 1).padStart(2, '0'); // Months are zero-indexed
+        const day = String(now.getUTCDate()).padStart(2, '0');
 
-		// return result
-		if (bookings.length === 0) {
-			return res
-				.status(StatusCodes.NOT_FOUND)
-				.send("booking not found");
-		}
-		return res.status(StatusCodes.OK).send(bookings);
-	},
+        return `${year}-${month}-${day}`;
+      };
+      // create new booking
+      const booking = await BookingModel.createBooking(
```

Comment:
why is the **Customers** object/info not passed to the **BookingModel**?

Reply:
Vì thêm customer vào booking sẽ ở 1 model khác